idp-binding-service/src/main/java/io/mosip/idp/binding/services/KeyBindingValidatorService.java

```diff
@@ -117,9 +124,9 @@ private boolean validateWLAToken(String individualId, String wlaToken, String fo
                     ConfigurableJWTProcessor jwtProcessor = new DefaultJWTProcessor();
                     jwtProcessor.setJWSKeySelector(keySelector);
                     jwtProcessor.setJWTClaimsSetVerifier(claimsSetVerifier);
-                    jwtProcessor.process(wlaToken, null); //If invalid throws exception
```

Comment:
In case of this code, if the header is not available, what was the behaviour ?

Reply:
JWSKeySelector will fail to select the key to verify the signature.
JWSKeySelector will fail in below cases:
1. if invalid sha256/sha1 thumbprint is provided.
2. if invalid keyid is provided.
3. if invalid alg/typ is provided

this is the exact exception
com.nimbusds.jose.proc.BadJOSEException: Signed JWT rejected: Another algorithm expected, or no matching key(s) found.

Reply:
My question was more of what if no keyid and no sha256 thumbprint is given in jwt ?

Reply:
if no keyid and no sha256 thumbprint is given in jwt,it just ignores the validation thats why I have explicitly added the check to confirm that it has sha256t header